src/Client.php

```diff
@@ -161,64 +158,32 @@ private function generateRandomString($state_length)
         return $state;
     }
 
-    /**
-     * Validate that the client_id and client_secret are the proper length.
-     *
-     * @param string $client_id                 The Client ID found in the admin panel
-     * @param string $client_secret             The Client Secret found in the admin panel
-     * @param string $api_host                  The api-host found in the admin panel
-     * @param string $redirect_url              The URL to redirect back to after the prompt
-     * @param boolean $use_duo_code_attribute   Flag to toggle returned code attribute name
-     *
-     * @return void
-     * @throws DuoException If parameters are not strings or for invalid Client ID or Client Secret
-     */
-    private function validateInitialConfig(
-        $client_id,
-        $client_secret,
-        $api_host,
-        $redirect_url,
-        $use_duo_code_attribute
-    ) {
-        if (!is_string($client_id) || !is_string($client_secret) || !is_string($api_host) || !is_string($redirect_url) || !is_bool($use_duo_code_attribute)
-        ) {
-            throw new DuoException(self::PARSING_CONFIG_ERROR);
```

Comment:
Relying on PHP type checking does change the behavior here - what used to be a (vague) DuoException will now be a TypeError if a value of the incorrect type is provided.  I am pondering whether that is a good plan; and if that makes this change backwards incompatible

Reply:
As a PHP developer the expectation is that a `TypeError` will be thrown for incorrect types - that's how the language and other modern libraries work.

Although virtually any change [could be considered a BC break](https://xkcd.com/1172/), it doesn't make sense why anyone would rely on the specific message thrown when passing a wrong type to these functions. They would just fix their code to pass the correct value.

Reply:
Seems reasonable to me.